src/main/java/mate/academy/internetshop/dao/jdbc/JdbcUserDao.java

```diff
@@ -0,0 +1,170 @@
+                return Optional.of(getUser(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public User create(User user) {
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO users (name, login, password) VALUES (?, ?, ?)";
+            PreparedStatement statement = conn.prepareStatement(query,
+                    PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            resultSet.next();
+            user.setId(resultSet.getLong(1));
+            saveRoles(user, conn);
+            return user;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users WHERE user_id = ?";
+            PreparedStatement statement = conn.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getUser(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users";
+            Statement statement = conn.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            List<User> users = new ArrayList<>();
+            while (resultSet.next()) {
+                users.add(getUser(resultSet));
+            }
+            return users;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            String query = "UPDATE users SET name = ?, login = ?, password = ? WHERE user_id = ?";
+            PreparedStatement statement = conn.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+            query = "DELETE FROM users_roles WHERE user_id = ?";
+            statement = conn.prepareStatement(query);
+            statement.setLong(1, user.getId());
+            saveRoles(user, conn);
```

Comment:
What if we decided to remove some roles from user and then update him?

Reply:
A missed executeUpdate, I understand.